test/framework/src/test/java/org/elasticsearch/test/VersionUtilsTests.java

```diff
@@ -0,0 +1,167 @@
+        }
+    }
+
+    public void testRandomVersionBetween() {
+        // full range
+        Version got = VersionUtils.randomVersionBetween(random(), VersionUtils.getFirstVersion(), Version.CURRENT);
+        assertTrue(got.onOrAfter(VersionUtils.getFirstVersion()));
+        assertTrue(got.onOrBefore(Version.CURRENT));
+        got = VersionUtils.randomVersionBetween(random(), null, Version.CURRENT);
+        assertTrue(got.onOrAfter(VersionUtils.getFirstVersion()));
+        assertTrue(got.onOrBefore(Version.CURRENT));
+        got = VersionUtils.randomVersionBetween(random(), VersionUtils.getFirstVersion(), null);
+        assertTrue(got.onOrAfter(VersionUtils.getFirstVersion()));
+        assertTrue(got.onOrBefore(Version.CURRENT));
+
+        // sub range
+        got = VersionUtils.randomVersionBetween(random(), Version.V_5_0_0,
+                Version.V_6_0_0_alpha2);
+        assertTrue(got.onOrAfter(Version.V_5_0_0));
+        assertTrue(got.onOrBefore(Version.V_6_0_0_alpha2));
+
+        // unbounded lower
+        got = VersionUtils.randomVersionBetween(random(), null, Version.V_6_0_0_alpha2);
+        assertTrue(got.onOrAfter(VersionUtils.getFirstVersion()));
+        assertTrue(got.onOrBefore(Version.V_6_0_0_alpha2));
+        got = VersionUtils.randomVersionBetween(random(), null, VersionUtils.allReleasedVersions().get(0));
+        assertTrue(got.onOrAfter(VersionUtils.getFirstVersion()));
+        assertTrue(got.onOrBefore(VersionUtils.allReleasedVersions().get(0)));
+
+        // unbounded upper
+        got = VersionUtils.randomVersionBetween(random(), Version.V_5_0_0, null);
+        assertTrue(got.onOrAfter(Version.V_5_0_0));
+        assertTrue(got.onOrBefore(Version.CURRENT));
+        got = VersionUtils.randomVersionBetween(random(), VersionUtils.getPreviousVersion(), null);
+        assertTrue(got.onOrAfter(VersionUtils.getPreviousVersion()));
+        assertTrue(got.onOrBefore(Version.CURRENT));
+
+        // range of one
+        got = VersionUtils.randomVersionBetween(random(), VersionUtils.getFirstVersion(), VersionUtils.getFirstVersion());
+        assertEquals(got, VersionUtils.getFirstVersion());
+        got = VersionUtils.randomVersionBetween(random(), Version.CURRENT, Version.CURRENT);
+        assertEquals(got, Version.CURRENT);
+        got = VersionUtils.randomVersionBetween(random(), Version.V_6_0_0_alpha2,
+                Version.V_6_0_0_alpha2);
+        assertEquals(got, Version.V_6_0_0_alpha2);
+
+        // implicit range of one
+        got = VersionUtils.randomVersionBetween(random(), null, VersionUtils.getFirstVersion());
+        assertEquals(got, VersionUtils.getFirstVersion());
+        got = VersionUtils.randomVersionBetween(random(), Version.CURRENT, null);
+        assertEquals(got, Version.CURRENT);
+    }
+
+    static class TestPatchBranch {
+        public static final Version V_5_3_0 = Version.fromId(Version.V_5_3_0_ID);
```

Comment:
Using constants from `Version` makes removing old constants time consuming. Can you use `Version.fromString` throughout these tests?